devito/symbolics/manipulation.py

```diff
@@ -146,6 +146,11 @@ def _(expr, args, kwargs):
 
 @_uxreplace_handle.register(Mul)
 def _(expr, args, kwargs):
+    # Perform some basic simplifications at least
+    args = [i for i in args if i != 1]
+    if any(i == 0 for i in args):
+        return S.Zero
```

Comment:
Can probably do `S.One` and `S.negativeOne` as well that have a bunch of special handling in sympy